app/controllers/friendship_acknowledgements_controller.rb

```diff
@@ -1,26 +1,37 @@
 class FriendshipAcknowledgementsController < ApplicationController
+  before_action :authenticate_user!
   def create
-    friend_request = FriendRequest.find(params[:friend_request][:friend_request_id])
-
-    ActiveRecord::Base.transaction do
-      friend_request.update(status: "accepted")
-      friendship1 = Friendship.create(user: friend_request.receiver, friend: friend_request.requester)
-      friendship2 = Friendship.create(user: friend_request.requester, friend: friend_request.receiver)
-      FriendshipAcknowledgement.create(friend_request_id: friend_request.id, friendship_id: [friendship1, friendship2])
-    end
-
-    redirect_to notifications_path
+    friend_request = find_friend_request
+    acknowledge_friend_request(friend_request, 'accepted')
   end
 
   def destroy
-    friend_request = FriendRequest.find(params[:friend_request][:friend_request_id])
+    friend_request = find_friend_request
+    acknowledge_friend_request(friend_request, 'rejected')
+  end
+
+  private
 
-    friend_request.update(status: "rejected")
+  def find_friend_request
+    FriendRequest.find(params[:friend_request][:friend_request_id])
+  end
 
+  def acknowledge_friend_request(friend_request, status)
+    ActiveRecord::Base.transaction do
+      friend_request.update!(status:)
+      create_friendships(friend_request) if status == 'accepted'
```

Comment:
This method runs creations that shall also use `create!` with the bang version. To denote that to the caller up here, I'd rename it too: 
```suggestion
      create_friendships!(friend_request) if status == 'accepted'
```